lib/registrations/registration_checker.rb

```diff
@@ -182,12 +182,21 @@ def contains_organizer_fields?(request, organizer_fields)
       def validate_update_status!(new_status, competition, current_user, target_user, registration, events)
         raise WcaExceptions::RegistrationError.new(:unprocessable_entity, Registrations::ErrorCodes::INVALID_REQUEST_DATA) unless
           Registration.competing_statuses.include?(new_status)
-        raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::COMPETITOR_LIMIT_REACHED) if
-          new_status == Registrations::Helper::STATUS_ACCEPTED && competition.competitor_limit_enabled? &&
-          competition.registrations.accepted.count >= competition.competitor_limit
         raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::ALREADY_REGISTERED_IN_SERIES) if
           new_status == Registrations::Helper::STATUS_ACCEPTED && existing_registration_in_series?(competition, target_user)
 
+        if new_status == Registrations::Helper::STATUS_ACCEPTED && competition.competitor_limit_enabled?
+          raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::COMPETITOR_LIMIT_REACHED) if
+            competition.registrations.competing_status_accepted.count >= competition.competitor_limit
+
+          if competition.enforce_newcomer_month_reservations? && !target_user.newcomer_month_eligible?
+            available_spots = competition.competitor_limit - competition.registrations.competing_status_accepted.count
+
+            raise WcaExceptions::RegistrationError.new(:forbidden, Registrations::ErrorCodes::NO_UNRESERVED_SPOTS_REMAINING) if
+              (available_spots <= competition.newcomer_month_reserved_spots_remaining) && competition.registration_currently_open?
```

Comment:
Why are you bothering to check `competition.registration_currently_open?` here, but not during the "general" competitor limit check above?

From my gut feeling, it should either be both or none.

Reply:
If registration is closed and the only people left for an organizer to accept are non-newcomers, we don't want the organizer's attempt to accept them being blocked by the newcomer limit. 

Although now that I've typed it out I'm not fully convinced - because it may well be the case that once registration is closed the organizer still wants to respect the newcomer month reserved spots